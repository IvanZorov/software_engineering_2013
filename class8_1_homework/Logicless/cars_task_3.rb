#/usr/bin/ruby

require 'csv'

list = File.open("car_sort.csv", "w")
list.write 'Varna'

if ARGV[0] == 'cars.csv'
    CSV.foreach(ARGV[0]) do |row|
            if ARGV[1].length == 1
                if row[0][0] == ARGV[1]
                    row[0][0] == ARGV[1]
                    list.puts
                    list.write row
                end
            elsif ARGV[1].length == 2
                if row[0][0..1] == ARGV[1]
                    row[0][0..1] == ARGV[1]
                    list.puts
                    list.write row
                end
            end
    end
else 
    puts 'ERROR ! Wrong csv file argument.'
end
list.close
