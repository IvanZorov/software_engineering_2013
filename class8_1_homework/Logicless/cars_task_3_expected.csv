Varna
["B3135ZX", "Renault", "Clio1", "1993"]
["B0909KA", "Renault", "Clio3", "2004"]
["B8435BF", "Trabant", "1", "1975"]
["B5432KL", "Bmw", "M3", "2011"]
["B3112MP", "Renault", "Clio1", "2012"]
["B3255AV", "Renault", "Clio3", "2004"]
["B7762KP", "Trabant", "2", "1975"]
["B6421OU", "Bmw", "M3", "2011"]
["B3454KH", "Renault", "Clio1", "2012"]
["B3495UU", "Renault", "Clio3", "2004"]
["B7590PP", "Trabant", "1", "1975"]
["B8424LZ", "Bmw", "M3", "2011"]
["B8712PT", "Renault", "Clio1", "2012"]
["B2345ZQ", "Renault", "Clio3", "2004"]
["B9933KS", "Trabant", "2", "1975"]
["B9654BM", "Bmw", "M3", "2011"]
["B3135MW", "Renault", "Clio1", "2012"]
["B8233MA", "Renault", "Clio3", "2004"]
["B9362IC", "Trabant", "1", "1975"]
["B0235ZX", "Bmw", "M3", "2011"]
["B9347OR", "Renault", "Clio1", "2012"]
["B9898LL", "Renault", "Clio3", "2004"]
["B5679BG", "Trabant", "2", "1975"]
["B2356RU", "Bmw", "M3", "2011"]